drivers/led/leds_group_multicolor.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2025 Seagate Technology LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT leds_group_multicolor
+
+/**
+ * @file
+ * @brief Driver for multi-color LED built from monochromatic LEDs.
+ */
+
+#include <zephyr/drivers/led.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(leds_group_multicolor, CONFIG_LED_LOG_LEVEL);
+
+struct leds_group_multicolor_config {
+	uint8_t num_leds;
+	const struct led_dt_spec *led;
+};
+
+static int leds_group_multicolor_set_color(const struct device *dev, uint32_t led,
+					   uint8_t num_colors, const uint8_t *color)
+{
+	const struct leds_group_multicolor_config *config = dev->config;
+
+	if (led != 0) {
+		return -EINVAL;
+	}
+	if (num_colors != config->num_leds) {
+		return -EINVAL;
+	}
+
+	for (uint8_t i = 0; i < num_colors; i++) {
+		int err;
+
+		err = led_set_brightness_dt(&config->led[i], color[i]);
+		if (err) {
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+static int leds_group_multicolor_init(const struct device *dev)
+{
+	const struct leds_group_multicolor_config *config = dev->config;
+
+	if (!config->num_leds) {
+		LOG_ERR("%s: no LEDs found (leds property missing ?)", dev->name);
+		return -ENODEV;
+	}
+
+	for (uint8_t i = 0; i < config->num_leds; i++) {
+		const struct led_dt_spec *led = &config->led[i];
+
+		if (!led_is_ready_dt(led)) {
+			LOG_ERR("%s: LED device %s is not ready", dev->name, led->dev->name);
+			return -ENODEV;
+		}
+	}
+
+	return 0;
+}
+
+static DEVICE_API(led, leds_group_multicolor_api) = {
+	.set_color	= leds_group_multicolor_set_color,
+};
+
+#define LED_DT_SPEC_GET_BY_PHANDLE_IDX(node_id, prop, idx)			\
+	LED_DT_SPEC_GET(DT_PHANDLE_BY_IDX(node_id, prop, idx))
+
+#define LEDS_GROUP_MULTICOLOR_DEVICE(inst)					\
+										\
+static const struct led_dt_spec led_group_multicolor_##inst[] = {		\
+	DT_INST_FOREACH_PROP_ELEM_SEP(						\
+			inst, leds, LED_DT_SPEC_GET_BY_PHANDLE_IDX, (,))	\
+};										\
+										\
+static const struct leds_group_multicolor_config				\
+			leds_group_multicolor_config_##inst = {			\
+	.num_leds	= ARRAY_SIZE(led_group_multicolor_##inst),		\
+	.led		= led_group_multicolor_##inst,				\
+};										\
+										\
+DEVICE_DT_INST_DEFINE(inst, &leds_group_multicolor_init, NULL,			\
+		      NULL, &leds_group_multicolor_config_##inst,		\
+		      POST_KERNEL, CONFIG_LED_INIT_PRIORITY,			\
+		      &leds_group_multicolor_api);
```

Comment:
I think this block should be intdented in one tab

Reply:
What is the rule here ? Some drivers add a tab, some don't.

Reply:
Ah, interesting, looks like `led_gpio.c` has it not indented. We refer back to the Linux coding style https://www.kernel.org/doc/html/v4.10/process/coding-style.html#macros-enums-and-rtl it's not quite super explicit about it, but the norm would be to indent it, most code does it, vim certainly does it, clang-format as well... is it really widespread? Can't see may examples but I see how this could slip in easily